e2e/template_test.go

```diff
@@ -0,0 +1,501 @@
+package e2e_test
+
+import "github.com/Azure/agentbaker/pkg/agent/datamodel"
+
+// this is huge, but accurate, so leave it here
+// TODO(ace): minimize the actual required defaults.
+// this is what we previously used for bash e2e from e2e/nodebootstrapping_template.json
+// which itself was extracted from baker_test.go logic, which was inherited from aks-engine.
+func baseTemplate() *datamodel.NodeBootstrappingConfiguration {
```

Comment:
How do we plan on testing with different NBCs: 1) mutate the result NBC of baseTemplate() when necessary, 2) create another function with a new NBC from scratch (will be quite large then) or some other way?

Reply:
If we go with option 1, then we need to make sure that other callers/dependencies of NBC all use the mutated NBC and not the base one by mistake.

Reply:
this is in fact done

I think we can also reduce the base template size, it has many defaults that aren't needed